contracts/libraries/WeightedOracleLibrary.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity >=0.5.0 <0.8.0;
+
+import '@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol';
+
+/// @title Weighted Oracle library
+/// @notice Provides functions to integrate with different tier oracles of the same V3 pair
+library WeightedOracleLibrary {
+
+    struct TimeWeightedObservation {
+        int24 timeWeightedAverageTick;
+        uint128 timeWeightedHarmonicAverageLiquidity;
+    }
+
+    /// @notice Fetches time-weighted observations across different Uniswap V3 pools
```

Comment:
i'd add a note indicating that pools should almost certainly be for the same underlying assets, and differ only in fee tier

Reply:
Done!

Could you check if it is clear enough now?